thirdParty/catch2/include/catch2/catch.hpp

```diff
@@ -11469,14 +11472,14 @@ namespace Floating {
 
         ret << " ([";
         if (m_type == FloatingPointKind::Double) {
-            write(ret, step(m_target, static_cast<double>(-INFINITY), m_ulps));
+            write(ret, step(m_target, -std::numeric_limits<double>::infinity(), m_ulps));
             ret << ", ";
-            write(ret, step(m_target, static_cast<double>( INFINITY), m_ulps));
+            write(ret, step(m_target,  std::numeric_limits<double>::infinity(), m_ulps));
         } else {
             // We have to cast INFINITY to float because of MinGW, see #1782
-            write(ret, step(static_cast<float>(m_target), static_cast<float>(-INFINITY), m_ulps));
-            ret << ", ";
-            write(ret, step(static_cast<float>(m_target), static_cast<float>( INFINITY), m_ulps));
+            write(ret, step(static_cast<float>(m_target), -std::numeric_limits<float>::infinity(), m_ulps));
+            ret << ", ";                                                                        
+            write(ret, step(static_cast<float>(m_target),  std::numeric_limits<float>::infinity(), m_ulps));
```

Comment:
These changes to catch2 are only required for some special cases:
* This one when compiling for offload to nvptx using clang.
* The other one with IMB's XL compiler.
Neither should be an issue in the CI. I will test this theory by pushing a version of this branch without these changes later.

These patches to not apply to the current catch2 master. I will have to open issues rather than a PR.